cabot/metricsapp/api/base.py

```diff
@@ -100,74 +86,102 @@ def _add_threshold_data(check, series):
+    threshold = check.high_alert_value
+    importance = check.high_alert_importance
+    return _points_trigger_alert(result, series, series_name, datapoints, check, threshold, importance)
+
+
+def _points_trigger_warning(result, series, series_name, datapoints, check):
+    threshold = check.warning_value
+    importance = Service.WARNING_STATUS
+    return _points_trigger_alert(result, series, series_name, datapoints, check, threshold, importance)
+
+
+def _points_trigger_alert(result, series, series_name, datapoints, check, threshold, importance):
+    '''
+    Process a set of points, looking to see if they breach the threshold for the
+    given alert level ('warn' or 'high'). If so, populate the result object and
+    return True. Otherwise return False.
+    '''
+    # If there's no threshold set, then ignore
+    if threshold is None:
+        return False
+
+    # Examine each data point
+    consecutive_failures = 0
+
+    for point in datapoints:
+        timestamp, value = point
+        # If the point fails, increment the consecutive failure count, and
+        # then check if we have enough failed points to return an error.
+        if _point_failure_check(check.check_type, threshold, value):
+            consecutive_failures += 1
+            if consecutive_failures >= check.consecutive_failures:
+                # Set the importance so the check fails at the right level
+                check.importance = importance
+                result.succeeded = False
+                result.error = _get_error_message(check, threshold, importance, series_name, value)
+                result.raw_data = _add_threshold_data(check, series)
+                return True
+        else:
+            consecutive_failures = 0
+
+    # No problems found
+    return False
+
+
 def run_metrics_check(check):
     """
     Run the status check.
     :param check: the status check
     :return: a StatusCheckResult containing success/failure/error information
     """
-    result = StatusCheckResult(check=check)
-    # result.succeeded will be set to False when failures happen
-    result.succeeded = True
-
+    # Get the series data. If there was an error, return immediately.
     series = check.get_series()
 
     if series['error'] is True:
-        result.succeeded = False
         message = series.get('error_message')
-        result.error = 'Error fetching metric from source: {}'.format(message)
         logger.exception('Error fetching metrics: {}: {}'.format(series.get('error_code'), message))
-        return result
+        error = 'Error fetching metric from source: {}'.format(message)
+        return StatusCheckResult(check=check, succeeded=False, error=error)
 
-    # Oldest point we'll look at (time range is in seconds)
-    earliest_point = time.time() - check.time_range * 60
+    # Ignore all checks before the following start time
+    start_time = time.time() - check.time_range * 60
+
+    def filter_old_points(p):
```

Comment:
Any specific reason to make this an inner function? feel like it could just as easily be a helper function

Reply:
Yeah, it references the `start_time` variable, defined outside the function.  Not sure if it'd be possible to get the same thing working with a helper function, as it needs that state.